### PR TITLE
Add doc reference for client_auth_method

### PR DESCRIPTION
**Pull request by jkakavas**
Support for additional Client authentication methods was added in
the OIDC realm in #58708. This change adds the `rp.client_auth_method`
in the realm settings reference doc.

### CODE REVIEW

**Review by ywangd — COMMENTED** [review]
Should we also document `rp.client_auth_jwt_signature_algorithm` while we are here?

File: docs/reference/settings/security-settings.asciidoc [hunk 1]
```diff
@@ -1633,6 +1633,23 @@ at the OpenID Connect Provider.
...
+(<<static-cluster-setting, Static>>)
+The Client authentication method used by elasticsearch as a client to authenticate
+to the OpenID Connect Provider. Can be `client_secret_basic`, `client_secret_post`,
+or `client_secret_jwt`. Defaults to `client_secret_basic`.
+// end::rp-client-auth-method-tag[]
+
+// tag::rp-client-auth-jwt-signature-algorithm[]
+`rp.client_auth_signature_algorithm` {ess-icon}::
+(<<static-cluster-setting, Static>>)
+The signature algorithm tat {es} uses to sign the JWT with which it authenticates
```
Comment (ywangd) [inline docs/reference/settings/security-settings.asciidoc:1647]:
```suggestion
The signature algorithm that {es} uses to sign the JWT with which it authenticates
```

**Review by ywangd — APPROVED** [review]
LGTM

File: docs/reference/settings/security-settings.asciidoc (hunk 1 above)
Comment (jkakavas) [reply docs/reference/settings/security-settings.asciidoc:1647]: :facepalm:

File: docs/reference/settings/security-settings.asciidoc [hunk 2]
```diff
@@ -1633,6 +1633,23 @@ at the OpenID Connect Provider.
 The OAuth 2.0 Client Secret that was assigned to {es} during registration
 at the OpenID Connect Provider.
 
+// tag::rp-client-auth-method-tag[]
+`rp.client_auth_method` {ess-icon}::
+(<<static-cluster-setting, Static>>)
+The Client authentication method used by elasticsearch as a client to authenticate
```
Comment (lcawl) [inline docs/reference/settings/security-settings.asciidoc:1639]: Feels one of the two occurrences of "client" is redundant:
```suggestion
The authentication method used by {es} as a client to authenticate
```

File: docs/reference/settings/security-settings.asciidoc [hunk 3]
```diff
@@ -1633,6 +1633,23 @@ at the OpenID Connect Provider.
...
+to the OpenID Connect Provider. Can be `client_secret_basic`, `client_secret_post`,
+or `client_secret_jwt`. Defaults to `client_secret_basic`.
+// end::rp-client-auth-method-tag[]
+
+// tag::rp-client-auth-jwt-signature-algorithm[]
+`rp.client_auth_signature_algorithm` {ess-icon}::
+(<<static-cluster-setting, Static>>)
+The signature algorithm that {es} uses to sign the JWT with which it authenticates
+as a Client to the OpenID Connect Provider when `client_secret_jwt` is selected for
```
Comment (lcawl) [inline docs/reference/settings/security-settings.asciidoc:1648]: Unless this is a specific client that is always capitalized, I think it should be lowercase:
```suggestion
as a client to the OpenID Connect Provider when `client_secret_jwt` is selected for
```

File: docs/reference/settings/security-settings.asciidoc (hunk 2 above)
Comment (jkakavas) [reply docs/reference/settings/security-settings.asciidoc:1639]: I'll change the {es} thanks ! but it's called "Client authentication" in OpenID Connect , so I will leave this here

**Pull request merged**